### ISSUE TITLE
Specify behavior when document is not fully active

### ISSUE THREAD

**Issue by inexorabletash** [w3c/IndexedDB#412]
Related to #381

1. navigate to https://example.com/ (or anything really)
1. open devtools/inspector/console/etc
1. `i = document.createElement('iframe');`
1. `i.src = location;`
1. `document.body.append(i);`
1. `ii = i.contentWindow.indexedDB;`
1. `ii.open('db');` - returns IDBOpenDBRequest
1. `i.remove();`
1. `ii.open('db');` - returns ???

In Chrome this returns null. Throwing is probably more correct behavior. I haven't tested in other browsers.

**Comment by asutherland** [comment]
[In Firefox we return our unknown error](https://searchfox.org/mozilla-central/rev/d81e60336d9f498ad3985491dc17c2b77969ade4/dom/indexedDB/IDBFactory.cpp#502-505) because we [cleared the global from the factory](https://searchfox.org/mozilla-central/rev/d81e60336d9f498ad3985491dc17c2b77969ade4/dom/indexedDB/IDBFactory.cpp#740-746) when the iframe's window was cleaned up.  In Firefox devtools this looks like:
> DOMException: The operation failed for reasons unrelated to the database itself and not covered by any other error code.

Throwing seems appropriate for a case like this where the API does not return a Promise.

**Comment by inexorabletash** [comment]
Agreed re: throwing.

For [web locks](https://www.w3.org/TR/web-locks/#dom-lockmanager-request) it's spec'd to reject with an "InvalidStateError" DOMException, which seems reasonable to me. I'd prefer to reserve UnknownError for cases when we're in unspecable/untestable territory, whereas this is just an edge case but totally specable/testable. I don't feel strongly about it though.

FWIW, in Chrome localStorage.getItem() throws a "SecurityError" DOMException and caches.open() returns undefined (not a Promise!). Wheee.

**Comment by asutherland** [comment]
Yes, InvalidStateError seems better and we're fine to change Firefox's behavior here (and have tests that enforce that as the right exception).  Thanks!

**Comment by attila-lendvai** [comment]
FWIW, i'm also getting this error with the MetaMask extension, and it completely fails to start now.

> UnknownError: The operation failed for reasons unrelated to the database itself and not covered by any other error code.

i have zero clue what goes wrong, where, and how to fix it. deleting the storage would mean hours of work lost.

**Comment by asutherland** [comment]
> FWIW, i'm also getting this error with the MetaMask extension, and it completely fails to start now.
> 
> > UnknownError: The operation failed for reasons unrelated to the database itself and not covered by any other error code.

This sounds browser specific.  If this is on Firefox, please [file a bug against our IndexedDB component](https://bugzilla.mozilla.org/enter_bug.cgi?component=Storage%3A+IndexedDB&product=Core) and I can respond there.  If this is on another browser, please file against that browser (or the MetaMask extension).
